bin/target_user.sh

```diff
@@ -8,29 +8,33 @@
 #
 log_src='['${0##*/}']'
 
+odoo_user='odoo'
+
 # Check if there's a target user to run Odoo
 if [ "$TARGET_ID" ]; then
-    echo $log_src[`date +%F.%H:%M:%S`]' TARGET_ID = '$TARGET_ID
-
     echo $log_src[`date +%F.%H:%M:%S`]' Check if target user exists...'
-    EXISTS=$( getent passwd $TARGET_ID | wc -l )
-    echo $log_src[`date +%F.%H:%M:%S`]' EXISTS = '$EXISTS
+    exists=$( getent passwd $TARGET_ID | wc -l )
 
     # Create target user
-    if [ $EXISTS == "0" ]; then
-        echo $log_src[`date +%F.%H:%M:%S`]' Create target user...'
-        adduser --uid $TARGET_ID --disabled-password --gecos "" --shell=/bin/bash target-user
+    if [ $exists == "0" ]; then
+        echo $log_src[`date +%F.%H:%M:%S`]' Create target Odoo user...'
+        odoo_user='target-odoo-user'
+        adduser --uid $TARGET_ID --disabled-password --gecos "" --shell=/bin/bash $odoo_user
 
         # Add target user to odoo group so that he can read/write the content of /opt/odoo
         echo $log_src[`date +%F.%H:%M:%S`]' Add target user to odoo group...'
-        usermod -a -G odoo target-user
-    # If the user already exists, check if it's the same as odoo
+        usermod -a -G odoo $odoo_user
     else
         echo $log_src[`date +%F.%H:%M:%S`]' Target user already exists, make sure it is odoo...'
-        ODOO_ID=$( id -u odoo )
-        if [ $TARGET_ID -ne $ODOO_ID ]; then
+        odoo_user_id=$( id -u $odoo_user )
+
+        # If the user already exists, check if it's the same as odoo
+        if [ $TARGET_ID -ne $odoo_user_id ]; then
             echo $log_src[`date +%F.%H:%M:%S`]' The target user is not the same as odoo, exiting'
```

Comment:
It is an unlikely scenario but I reckon still possible.
Actually another possible output would be to switch the existing target user and Odoo user. It might potentially have side effects (the existing user id is used in another process/batch) but unlikely.
Still your solution is safe so I am ok with it (annoying if you cannot change the user id in the host). 
IMO: the container has potentially fewer users and possible conflicts than the host so switching the users in the container might be the simplest over time)  
I would just suggest at least to improve the exit message such as:
"The id selected for target user is used by another user ($user_name) in the container: exiting"